lib/jnpr/junos/utils/sw.py

```diff
@@ -991,29 +1005,44 @@ def reboot(self, in_min=0, at=None, all_re=True):
     # poweroff - system shutdown
     # -------------------------------------------------------------------------
 
-    def poweroff(self, in_min=0):
+    def poweroff(self, in_min=0, on_node=None):
         """
         Perform a system shutdown, with optional delay (in minutes) .
 
         If the device is equipped with dual-RE, then both RE will be
         rebooted.  This code also handles EX/QFX VC.
 
-        :param int in_min: time (minutes) before rebooting the device.
+        :param int in_min: time (minutes) before shutting down the device.
+
+        :param str on_node: In case of linux based device, function will by default
+            shutdown the whole device. If any specific node is mentioned,
+            shutdown will be performed on mentioned node
 
         :returns:
-            * reboot message (string) if command successful
+            * power-off message (string) if command successful
 
         :raises RpcError: when command is not successful.
 
         .. todo:: need to better handle the exception event.
         """
-        cmd = E('request-power-off', E('in', str(in_min)))
-
-        if self._multi_RE is True and self._multi_VC is False:
-            cmd.append(E('both-routing-engines'))
+        if self._dev.facts['_is_linux']:
+            if on_node is None:
+                cmd = E('request-shutdown-power-off')
+            else:
+                cmd = E('request-node-power-off')
+                cmd.append(E('node', on_node))
+        else:
+            cmd = E('request-power-off')
+            if self._multi_RE is True and self._multi_VC is False:
+                cmd.append(E('both-routing-engines'))
+        cmd.append(E('in', str(in_min)))
         try:
             rsp = self.rpc(cmd)
-            return rsp.getparent().findtext('.//request-reboot-status').strip()
+            if self._dev.facts['_is_linux']:
+                got = rsp.text
+            else:
+                got = rsp.getparent().findtext('.//request-reboot-status').strip()
```

Comment:
@rsmekala Have you tested that for this new case too rpc reply will have request-reboot-status tag?